compiler+runtime/test/cpp/jank/read/lex.cpp

```diff
@@ -460,17 +478,332 @@ namespace jank::read::lex
+        native_vector<result<token, error>> const tokens(p.begin(), p.end());
         CHECK(tokens
               == make_tokens({
                 { 0, 4, token_kind::integer, 1234ll }
         }));
       }
 
+      SUBCASE("Positive multiple numbers")
+      {
+        processor p{ "0 1234" };
+        native_vector<result<token, error>> const tokens(p.begin(), p.end());
+        CHECK(tokens
+              == make_tokens({
+                { 0, 1, token_kind::integer,    0ll },
+                { 2, 4, token_kind::integer, 1234ll },
+        }));
+      }
+
+      SUBCASE("Octal number")
+      {
+        processor p{ "034 -034 08.9 07e1" };
+        native_vector<result<token, error>> const tokens(p.begin(), p.end());
+        CHECK(tokens
+              == make_tokens({
+                {  0, 3, token_kind::integer,  28ll },
+                {  4, 4, token_kind::integer, -28ll },
+                {  9, 4,    token_kind::real,   8.9 },
+                { 14, 4,    token_kind::real,  70.0 },
+        }));
+      }
+
+      SUBCASE("Invalid octal number")
+      {
+        processor p{ "08 0a -08" };
+        native_vector<result<token, error>> const tokens(p.begin(), p.end());
+        CHECK(tokens
+              == make_results({
+                error{ 0, 2, "invalid number: char 8 are invalid for radix 8" },
+                error{ 3, 5, "invalid number: char a are invalid for radix 8" },
+                error{ 6, 9, "invalid number: char 8 are invalid for radix 8" },
+        }));
+      }
+
+      SUBCASE("Octal numbers with invalid padding")
+      {
+        processor p{ "000123 0123 034 00" };
+        native_vector<result<token, error>> const tokens(p.begin(), p.end());
+        CHECK(tokens
+              == make_tokens({
+                {  0, 6, token_kind::integer, 83ll },
+                {  7, 4, token_kind::integer, 83ll },
+                { 12, 3, token_kind::integer, 28ll },
+                { 16, 2, token_kind::integer,  0ll }
+        }));
+      }
+
+      SUBCASE("Edge case: Invalid mixed radix")
+      {
+        processor p{ "123abc 0x12g 8r8 16rx.2" };
+        native_vector<result<token, error>> const tokens(p.begin(), p.end());
+        CHECK(tokens
+              == make_results({
+                error{ 0, 6, "invalid number: char abc are invalid for radix 10" },
+                error{ 7, 12, "invalid number: char g are invalid for radix 16" },
+                error{ 13, 16, "invalid number: char 8 are invalid for radix 8" },
+                error{ 17, 21, "arbitrary radix number can only be an integer" },
+                error{ 21, 21, "unexpected character: ." },
+                error{ 22, 22, "expected whitespace before next token" },
+                token{ 22, 1, token_kind::integer, 2ll },
+        }));
+      }
+
+      SUBCASE("Arbitrary radix edge cases")
+      {
+        /* exceeds 64-bit integer max */
+        processor p{ "36r0123456789abcdefghijklmnopqrstuvwxyz" };
```

Comment:
Seems like this should be an error, until we support big numbers. The token we create is lossy.

Reply:
updated to an error instead.